File: app/src/main/java/net/afterday/compas/persistency/items/ItemsPersistency.java
package net.afterday.compas.persistency.items;

import java.util.List;
import java.util.Map;

/**
 * Created by Justas Spakauskas on 2/10/2018.
 */

public interface ItemsPersistency
{
    Map<String, ItemDescriptor> getItemsByCode();
    ItemDescriptor getItemForCode(String code);
    Map<Integer, List<ItemDescriptor>> getItemsAddeWithLevel();
}

File: engine/src/main/kotlin/net/afterday/compas/engine/persistency/items/ItemsPersistency.kt
package net.afterday.compas.engine.persistency.items

import net.afterday.compas.engine.core.inventory.items.ItemDescriptor

interface ItemsPersistency {
    fun getItemsByCode(): Map<String, ItemDescriptor>?
    fun getItemForCode(code: String): ItemDescriptor?
    fun getItemsAddeWithLevel(): Map<Int, List<ItemDescriptor>>?
}